src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityPumpHatch.java

```diff
@@ -0,0 +1,141 @@
+package gregtech.common.metatileentities.multi;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.capability.impl.FluidTankList;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.FluidContainerSlotWidget;
+import gregtech.api.gui.widgets.ImageWidget;
+import gregtech.api.gui.widgets.SlotWidget;
+import gregtech.api.gui.widgets.TankWidget;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockAbilityPart;
+import gregtech.api.metatileentity.multiblock.MultiblockAbility;
+import gregtech.api.render.ICubeRenderer;
+import gregtech.api.render.Textures;
+import gregtech.common.metatileentities.electric.multiblockpart.MetaTileEntityMultiblockPart;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidTank;
+import net.minecraftforge.fluids.IFluidTank;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class MetaTileEntityPumpHatch extends MetaTileEntityMultiblockPart implements IMultiblockAbilityPart<IFluidTank> {
+
+    private static final int FLUID_TANK_SIZE = 1000;
+    private final FluidTank waterTank;
+    private final ItemStackHandler containerInventory;
+
+    public MetaTileEntityPumpHatch(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId, 0);
+        this.waterTank = new FluidTank(FLUID_TANK_SIZE);
+        this.containerInventory = new ItemStackHandler(2);
+        initializeInventory();
```

Comment:
At the end of the constructor: add the following to prevent the hatch and only the hatch from being tinted: `this.setPaintingColor(0xFFFFFF)`. It prevents weird color differences that appear when only one part of a multiblock structure is tinted. It's only a visual change, but it looks weird when tinting.